go/arrow/csv/reader.go

```diff
@@ -247,12 +247,8 @@ func (r *Reader) validate(recs []string) {
 }
 
 func (r *Reader) isNull(val string) bool {
-	for _, ns := range r.nullValues {
-		if val == ns {
-			return true
-		}
-	}
-	return false
+	_, ok := r.nulls[val]
```

Comment:
i commented on the other PR, but I would expect this to potentially have worse performance for small lists if the map doesn't have explicit optimizations for small collection sizes.

Reply:
good point.